### PR TITLE
nxos_interfaces model

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

After build:

**DOCSTRING:**
```
#!/usr/bin/python
# -*- coding: utf-8 -*-
# Copyright 2019 Red Hat
# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)

##############################################
################# WARNING ####################
##############################################
###
### This file is auto generated by the resource
###   module builder playbook.
###
### Do not edit this file manually.
###
### Changes to this file will be over written
###   by the resource module builder.
###
### Changes should be made in the model used to
###   generate this file or in the resource module
###   builder template.
###
##############################################
##############################################
##############################################

"""
The module file for nxos_interfaces
"""

from __future__ import absolute_import, division, print_function
from ansible.module_utils.basic import AnsibleModule
from ansible_collections.ansible_network.network.plugins.module_utils. \
     nxos.config.interfaces.interfaces import Interfaces

ANSIBLE_METADATA = {'metadata_version': '1.1',
                    'status': [u'preview'],
                    'supported_by': [u'Ansible Network']}


DOCUMENTATION = """
---
module: nxos_interfaces
version_added: 2.9
short_description: Manages <xxxx> attributes of <network_os> <resource>.
description: [u'Manages <xxxx> attributes of <network_os> <resource>.']
author: [u'Trishna Guha (@trishnaguha)']
options:
  config:
    description: The provided configuration
    suboptions:
      description:
        description:
        - Interface description.
        type: str
      duplex:
        default: auto
        description:
        - Interface link status. Applicable for ethernet interface only.
        type: str
      enable:
        default: true
        description:
        - Administrative State of interface.
        type: bool
      fabric_forwarding_anycast_gateway:
        description:
        - Associate SVI with anycast gateway under VLAN configuration mode. Applicable
          for SVI interface only.
        type: bool
      ip_forward:
        description:
        - Enable/Disable ip forward feature on SVIs.
        type: bool
      mode:
        description:
        - Manage Layer2 or Layer3 state of the interface. Applicable for Ethernet
          and portchannel interface only.
        type: str
      mtu:
        description:
        - MTU for a specific interface. Must be an even number between 576 and 9216.
          Applicable for Ethernet interface only.
        type: str
      name:
        description:
        - Full name of interface, i.e. Ethernet1/1, port-channel10.
        type: str
      speed:
        description:
        - Interface link speed. Applicable for ethernet interface only.
        type: str
  state:
    choices:
    - merged
    - replaced
    - overridden
    - deleted
    default: merged
    description:
    - The state the configuration should be left in
    type: str

"""

EXAMPLES = """
---

# Using merged

- name: Configure Eth interfaces
  nxos_interfaces:
    config:
      - name: Ethernet1/1
        description: 'Configured by Ansible'
        enable: True
      - name: Ethernet1/2
        description: 'Configured by Ansible Network'
        enable: False
	mode: layer3
    operation: merged

# Using replaced

- name: Configure mentioned interfaces and replace existing config
  nxos_interfaces:
    config:
      - name: Ethernet1/1
        description: 'Configured by Ansible'
        enable: True
	mtu: 2000
      - name: Ethernet1/2
        description: 'Configured by Ansible Network'
        enable: False
	mode: layer2
    operation: replaced

# Using overridden

- name: Override interfaces
  nxos_interfaces:
    config:
      - name: Ethernet1/1
        description: 'Configured by Ansible'
        enable: True
	mode: layer3
      - name: Ethernet1/2
        description: 'Configured by Ansible Network'
        enable: False
	speed: 1000
    operation: overridden

# Using deleted

- name: Delete given arguments for the interface
  nxos_interfaces:
    config:
      - name: Ethernet1/1
        description: 'Configured by Ansible'
      - name: Ethernet1/2
        description: 'Configured by Ansible Network'
        mtu: 1800
    operation: deleted
"""
```

**LAYOUT:**
```
plugins/
├── action
├── filter
├── inventory
├── modules
│   ├── __init__.py
│   ├── nxos_facts.py
│   ├── nxos_interfaces.py
│   └── tags
└── module_utils
    ├── __init__.py
    └── nxos
        ├── argspec
        │   ├── facts
        │   │   ├── facts.py
        │   │   └── __init__.py
        │   ├── __init__.py
        │   └── interfaces
        │       ├── __init__.py
        │       └── interfaces.py
        ├── config
        │   ├── base.py
        │   ├── __init__.py
        │   └── interfaces
        │       ├── __init__.py
        │       └── interfaces.py
        ├── facts
        │   ├── base.py
        │   ├── facts.py
        │   ├── __init__.py
        │   └── interfaces
        │       ├── __init__.py
        │       └── interfaces.py
        ├── __init__.py
        └── utils
            ├── __init__.py
            └── utils.py
```

### CODE REVIEW

File: models/nxos/interfaces/nxos_interfaces.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  resource: interfaces
+  version_added: 2.9
+  short_description: 'Manages Interfaces attributes of NX-OS Interfaces.'
+  description:
+  - Manages Interfaces attributes of NX-OS Interfaces.
+  authors:
+  - Trishna Guha (@trishnaguha)
+  extends_documentation_fragment: nxos
+  notes:
+    - Tested against <network_os> 7.3.(0)D1(1) on VIRL
```
Comment (NilashishC) [inline models/nxos/interfaces/nxos_interfaces.yml]: <network_os> needs to be replaced with NX-OS

File: models/nxos/interfaces/nxos_interfaces.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
+generator_version: 1.0
+metadata:
+  vesion: 1.1
```
Comment (NilashishC) [inline models/nxos/interfaces/nxos_interfaces.yml]:
```suggestion
  version: 1.1
```

File: models/nxos/interfaces/replaced_example_01.txt [hunk 3]
```diff
@@ -0,0 +1,15 @@
+
+# Using replaced
+
+- name: Configure mentioned interfaces and replace existing config
+  nxos_interfaces:
+    config:
+      - name: Ethernet1/1
+        description: 'Configured by Ansible'
+        enable: True
+	mtu: 2000
```
Comment (NilashishC) [inline models/nxos/interfaces/replaced_example_01.txt]: Minor indentation issue here?

File: models/nxos/interfaces/replaced_example_01.txt [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      - name: Ethernet1/2
+        description: 'Configured by Ansible Network'
+        enable: False
+	mode: layer2
```
Comment (NilashishC) [inline models/nxos/interfaces/replaced_example_01.txt]: Same as above.

File: models/nxos/interfaces/replaced_example_01.txt (hunk 3 above)
Comment (trishnaguha) [reply models/nxos/interfaces/replaced_example_01.txt]: Looks like git issue? Weird that it neither shows up on my editor nor if you view the file here including the raw version https://raw.githubusercontent.com/ansible-network/resource_module_builder/f322a357d29267b62df374bd6d2019e01274bc7c/models/nxos/interfaces/replaced_example_01.txt.

File: models/nxos/interfaces/replaced_example_01.txt (hunk 4 above)
Comment (trishnaguha) [reply models/nxos/interfaces/replaced_example_01.txt]: same as above https://github.com/ansible-network/resource_module_builder/pull/15/files#r271160670

**Comment by trishnaguha** [comment]
Moved to https://github.com/ansible-network/resource_module_models/pull/3

**Pull request closed without merging**